File: 771-Jewels and Stones.py
# 36ms
class Solution:
    def numJewelsInStones(self, J, S):
        """
        :type J: str
        :type S: str
        :rtype: int
        """
        return sum([x in J for x in S])  # 会生成bool列表，然后求和


'''
# 暴力的优化 | 40ms
class Solution:
    def numJewelsInStones(self, J, S):
        """
        :type J: str
        :type S: str
        :rtype: int
        """
        n = 0
        for j in J:
            n += S.count(j)
        return n
'''


'''
# 暴力 | 52ms
class Solution:
    def numJewelsInStones(self, J, S):
        """
        :type J: str
        :type S: str
        :rtype: int
        """
        n = 0
        for j in J:
            for s in S:
                if j == s: n += 1
        return n
'''

File: C++/771-Jewels_and_Stones.cpp
// Runtime: 0 ms, faster than 100.00% of C++ online submissions for Jewels and Stones.
// Memory Usage: 8.4 MB, less than 59.15% of C++ online submissions for Jewels and Stones.
class Solution {
public:
    int numJewelsInStones(string J, string S) {
        int res = 0;
        for (char s : S)
            for (char j : J)
                if (s == j)
                    res++;
        return res;
    }
};